pkg/security/security_profile/dump/local_storage.go

```diff
@@ -58,19 +61,34 @@ func (s dumpFilesSlice) Less(i, j int) bool {
 
 // ActivityDumpLocalStorage is used to manage ActivityDumps storage
 type ActivityDumpLocalStorage struct {
-	localDumps *simplelru.LRU[string, *[]string]
+	sync.Mutex
+	deletedCount *atomic.Uint64
+	localDumps   *simplelru.LRU[string, *[]string]
 }
 
 // NewActivityDumpLocalStorage creates a new ActivityDumpLocalStorage instance
-func NewActivityDumpLocalStorage(cfg *config.Config) (ActivityDumpStorage, error) {
-	lru, err := simplelru.NewLRU(cfg.RuntimeSecurity.ActivityDumpLocalStorageMaxDumpsCount, func(name string, files *[]string) {
+func NewActivityDumpLocalStorage(cfg *config.Config, m *ActivityDumpManager) (ActivityDumpStorage, error) {
+	adls := &ActivityDumpLocalStorage{
+		deletedCount: atomic.NewUint64(0),
+	}
+
+	var err error
+	adls.localDumps, err = simplelru.NewLRU(cfg.RuntimeSecurity.ActivityDumpLocalStorageMaxDumpsCount, func(name string, files *[]string) {
 		if len(*files) == 0 {
 			return
 		}
+
+		// notify the security profile directory provider that we're about to delete a profile
+		if m.securityProfileManager != nil {
+			m.securityProfileManager.OnLocalStorageCleanup(*files)
+		}
+
 		// remove everything
 		for _, f := range *files {
 			_ = os.Remove(f)
 		}
+
+		adls.deletedCount.Add(1)
```

Comment:
Should we count the number of files deleted instead ? Like `.Add(len(*files))` ? (Even if, it may be 1 per 1, it's strange to have a loop and a single increment)

Reply:
I hesitated, but if we counted the number of files deleted, then it wouldn't match with the count of entries in `localDumps` (see how activity_dump.local_storage.count is generated) - which would make things even harder to debug. I updated the comment explaining what the metric counts to make it clearer.